### PR TITLE
[FAR-830] refactor usage rules

### PR DESCRIPTION
**Pull request by MartinBelthle**
**Please check if the PR fulfills these requirements** 
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?**

No.

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

This PR does 3 things:
1) The UsageMethod `TO_BE_EVALUATED` does no longer exist. It has been replaced by `AVAILABLE` (or `FORCED` for `Instant.AUTO`) and checked every time as it used to be.

2) UsageMethod `FORCED` now prevails over `AVAILABLE`

3) Remedial Actions with UsageMethod `FORCED` are no longer considered available, except for `AUTO` instant.

4) Modification of crac version 1.7 : `OnFlowConstraint`, `OnFlowConstraintInCountry`, `OnAngleConstraint` and `OnVoltageConstraint` usage rules now require a field usageMethod, see example below:

**Before v1.7**
``` java
"onFlowConstraintUsageRules" : [ {
  "instant" : "preventive",
  "flowCnecId" : "cnec3prevId"
} ]
```

**After v1.7**
``` java
"onFlowConstraintUsageRules" : [ {
  "instant" : "preventive",
  "flowCnecId" : "cnec3prevId",
  "usageMethod" : "available"
} ]
```
To ensure retrocompatibility, with older versions where the usageMethod is not specified, it is set to `AVAILABLE` (or `FORCED` for `Instant.AUTO`)

**What is the current behavior?** *(You can also link to an open issue here)*

1) The usageRules `OnFlowConstraint` and `OnFlowConstraintInCountry` return the usageMethod `TO_BE_EVALUATED`  when given the right state.
2)  We have a remedial action:
``` 
"id" : "PST_1",
    "name" : "CRA_1",
    "freeToUseUsageRules" : [ {
      "instant" : "curative",
      "usageMethod" : "available"
    } ],
   "onContingencyStateUsageRules" : [ {
      "instant" : "curative",
      "contingencyId" : "contingency1",
      "usageMethod" : "forced"
    } ]
```
This remedial action will be `AVAILABLE` for every curative state.
3) We have the same as above but for Instant AUTO.
The remedial action will never be usable.
4) See previous section.

**What is the new behavior (if this is a feature change)?**

1) The usageRules `OnFlowConstraint` and `OnFlowConstraintInCountry` now return `AVAILABLE` usageMethod (or `FORCED` for `Instant.AUTO`) when given the right state.
2) It will be  `AVAILABLE` for every curative state except for "contingency1" where it will not be considered.
3) It would be `FORCED` for "contingency1" and not considered for other states.
4) See first section.

**Does this PR introduce a breaking change?**

No.

**Other information**:

Every test passed with cucumber branch `remove_to_be_evaluated`

The website doc has been modified with this PR : https://github.com/farao-community/farao-community.github.io/pull/74

### CODE REVIEW

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 1]
```diff
@@ -200,10 +197,10 @@ TopoAutomatonSimulationResult simulateTopologicalAutomatons(State automatonState
         // -- First get forced network actions
         Set<NetworkAction> appliedNetworkActions = crac.getNetworkActions(automatonState, UsageMethod.FORCED);
 
-        // -- Then add those with TO_BE_EVALUATED usage method when evaluation condition is verified
+        // -- Then add those with AVAILABLE usage method when evaluation condition is verified
         // -- Evaluation condition is isAnyMarginNegative amongst network actions' flow cnecs associated to their usage rules
-        crac.getNetworkActions(automatonState, UsageMethod.TO_BE_EVALUATED).stream()
-                .filter(na -> na.isRemedialActionAvailable(automatonState, RaoUtil.isAnyMarginNegative(prePerimeterSensitivityOutput, na.getFlowCnecsConstrainingUsageRules(crac.getFlowCnecs(), network, automatonState), raoParameters.getObjectiveFunctionParameters().getType().getUnit())))
+        crac.getNetworkActions(automatonState, UsageMethod.AVAILABLE).stream()
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:202]: except OnState(AVAILABLE) and OnInstant(AVAILABLE)

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 2]
```diff
@@ -312,13 +309,13 @@ RangeAutomatonSimulationResult simulateRangeAutomatons(State automatonState, Sta
     Set<FlowCnec> gatherFlowCnecsForAutoRangeAction(RangeAction<?> availableRa,
                                                     State automatonState,
                                                     Network network) {
-        // UsageMethod is either FORCED or TO_BE_EVALUATED
+        // UsageMethod is either FORCED or AVAILABLE
         if (availableRa.getUsageMethod(automatonState).equals(UsageMethod.FORCED)) {
             return crac.getFlowCnecs(automatonState);
-        } else if (availableRa.getUsageMethod(automatonState).equals(UsageMethod.TO_BE_EVALUATED)) {
+        } else if (availableRa.getUsageMethod(automatonState).equals(UsageMethod.AVAILABLE)) {
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:315]: AND it is onFlowConstraint or OnFlowConstraintInCountry

File: ra-optimisation/search-tree-rao/src/test/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulatorTest.java [hunk 3]
```diff
@@ -243,7 +243,7 @@ void testGatherCnecs() {
     @Test
     void testGatherCnecsError() {
         RangeAction<?> ra = Mockito.mock(RangeAction.class);
-        Mockito.when(ra.getUsageMethod(autoState)).thenReturn(UsageMethod.AVAILABLE);
+        Mockito.when(ra.getUsageMethod(autoState)).thenReturn(UsageMethod.UNDEFINED);
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/test/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulatorTest.java:246]: you'll have to do something more complex here

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 4]
```diff
@@ -88,8 +90,8 @@ public AutomatonSimulator(Crac crac, RaoParameters raoParameters, ToolProvider t
      */
     AutomatonPerimeterResultImpl simulateAutomatonState(State automatonState, State curativeState, Network network) {
         TECHNICAL_LOGS.info("Optimizing automaton state {}.", automatonState.getId());
-        if (!crac.getNetworkActions(automatonState, UsageMethod.AVAILABLE).isEmpty()) {
-            BUSINESS_WARNS.warn("CRAC has network action automatons with usage method AVAILABLE. These are not supported.");
+        if (doesCracContainsUnsupportedRemedialActions(automatonState)) {
+            BUSINESS_WARNS.warn("CRAC has remedial action automatons with usage rule OnState(AVAILABLE) or OnInstant(AVAILABLE). These usage rules are not supported for this instant and will be ignored.");
         }
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:95]: you didn't have to change this behaviour

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 5]
```diff
@@ -142,8 +144,15 @@ private PrePerimeterSensitivityAnalysis getPreAutoPerimeterSensitivityAnalysis(S
         Set<FlowCnec> flowCnecsInSensi = crac.getFlowCnecs(automatonState);
         flowCnecsInSensi.addAll(crac.getFlowCnecs(curativeState));
         Set<RangeAction<?>> rangeActionsInSensi = new HashSet<>();
-        rangeActionsInSensi.addAll(crac.getRangeActions(automatonState, UsageMethod.FORCED, UsageMethod.TO_BE_EVALUATED));
-        rangeActionsInSensi.addAll(crac.getRangeActions(curativeState, UsageMethod.AVAILABLE, UsageMethod.FORCED, UsageMethod.TO_BE_EVALUATED));
+
+        // For automatonState, filters out rangeActions with usageMethod AVAILABLE having only
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:148]: we should only support FORCED automatons

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 6]
```diff
@@ -200,10 +209,10 @@ TopoAutomatonSimulationResult simulateTopologicalAutomatons(State automatonState
         // -- First get forced network actions
         Set<NetworkAction> appliedNetworkActions = crac.getNetworkActions(automatonState, UsageMethod.FORCED);
 
-        // -- Then add those with TO_BE_EVALUATED usage method when evaluation condition is verified
+        // -- Then add those with AVAILABLE usage method when evaluation condition is verified
         // -- Evaluation condition is isAnyMarginNegative amongst network actions' flow cnecs associated to their usage rules
-        crac.getNetworkActions(automatonState, UsageMethod.TO_BE_EVALUATED).stream()
-                .filter(na -> na.isRemedialActionAvailable(automatonState, RaoUtil.isAnyMarginNegative(prePerimeterSensitivityOutput, na.getFlowCnecsConstrainingUsageRules(crac.getFlowCnecs(), network, automatonState), raoParameters.getObjectiveFunctionParameters().getType().getUnit())))
+        crac.getNetworkActions(automatonState, UsageMethod.AVAILABLE).stream()
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:214]: we should only support FORCED automatons

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 7]
```diff
@@ -312,13 +321,18 @@ RangeAutomatonSimulationResult simulateRangeAutomatons(State automatonState, Sta
     Set<FlowCnec> gatherFlowCnecsForAutoRangeAction(RangeAction<?> availableRa,
                                                     State automatonState,
                                                     Network network) {
-        // UsageMethod is either FORCED or TO_BE_EVALUATED
+        // UsageMethod is either FORCED or AVAILABLE
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:324]: we should only support FORCED automatons

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 8]
```diff
@@ -329,10 +343,10 @@ List<List<RangeAction<?>>> buildRangeActionsGroupsOrderedBySpeed(PrePerimeterRes
         // 1) Get available range actions
         // -- First get forced range actions
         Set<RangeAction<?>> availableRangeActions = crac.getRangeActions(automatonState, UsageMethod.FORCED);
-        // -- Then add those with TO_BE_EVALUATED usage method when evaluation condition is verified
-        // -- Evaluation condition is isAnyMarginNegative amongst network actions' flow cnecs associated to their usage rules
-        crac.getRangeActions(automatonState, UsageMethod.TO_BE_EVALUATED).stream()
-            .filter(na -> na.isRemedialActionAvailable(automatonState, RaoUtil.isAnyMarginNegative(rangeActionSensitivity, na.getFlowCnecsConstrainingUsageRules(crac.getFlowCnecs(), network, automatonState), raoParameters.getObjectiveFunctionParameters().getType().getUnit())))
+        // -- Then add those with AVAILABLE usage method when evaluation condition is verified
+        // -- Evaluation condition is isAnyMarginNegative amongst range actions flow cnecs associated to their usage rules
+        crac.getRangeActions(automatonState, UsageMethod.AVAILABLE).stream()
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:348]: we should only support FORCED automatons

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/OnFlowConstraintImpl.java [hunk 9]
```diff
@@ -20,7 +20,7 @@ public class OnFlowConstraintImpl extends AbstractUsageRule implements OnFlowCon
     private FlowCnec flowCnec;
 
     OnFlowConstraintImpl(Instant instant, FlowCnec flowCnec) {
-        super(UsageMethod.TO_BE_EVALUATED);
+        super(UsageMethod.AVAILABLE);
```
Comment (pet-mit) [inline data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/OnFlowConstraintImpl.java:23]: actually you have to add this as an argument + attribute, and take it from the Adder. Same or OnFlowConstraintInCountry, OnVoltageContraint, OnAngleConstraint

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java [hunk 10]
```diff
@@ -391,11 +389,6 @@ static boolean checkAlignedRangeActions(State automatonState, List<RangeAction<?
             BUSINESS_WARNS.warn("Range action group {} contains range actions of different types; they are not simulated", alignedRa.get(0).getGroupId().orElseThrow());
             return false;
         }
-        // Ignore aligned range actions when one element of the group has a different usage method than the others
-        if (alignedRa.stream().map(rangeAction -> rangeAction.getUsageMethod(automatonState)).distinct().count() > 1) {
-            BUSINESS_WARNS.warn("Range action group {} contains range actions with different usage methods; they are not simulated", alignedRa.get(0).getGroupId().orElseThrow());
-            return false;
-        }
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:398]: why did you remove this?

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/commons/RaoUtil.java [hunk 11]
```diff
@@ -128,6 +125,39 @@ public static boolean isAnyMarginNegative(FlowResult flowResult, Set<FlowCnec> f
...
+        if (state.getInstant().equals(Instant.AUTO)) {
+            if (usageRulesOnTime.stream().anyMatch(usageRule -> usageRule.getUsageMethod(state).equals(UsageMethod.FORCED))) {
+                return true;
+            }
+        } else if (usageRulesOnTime.stream()
+            .anyMatch(usageRule -> usageRule.getUsageMethod(state).equals(UsageMethod.AVAILABLE)
+                || usageRule.getUsageMethod(state).equals(UsageMethod.FORCED))) {
+            return true;
+        }
+        return remedialAction.isRemedialActionAvailable(state, isAnyMarginNegative(prePerimeterResult, remedialAction.getFlowCnecsConstrainingUsageRules(flowCnecs, network, state), raoParameters.getObjectiveFunctionParameters().getType().getUnit()));
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/commons/RaoUtil.java:158]: I think there's a case not covered by this logic : if a RA has an OnInstant AVAILABLE usage rule, and another OnFlowConstraint UNAVAILABLE usage rule. If the CNEC is constrained, the RA should not be available.

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java (hunk 10 above)
Comment (MartinBelthle) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java:398]: Before using this method (only once in the code) we select only available ra (meaning with usageMethod FORCED and with flow constraint for ofc usageRules). Before we used to have FORCED and TO_BE_EVALUATED ones but not now so this check seems useless to me.
``` java
// -- First get forced range actions
Set<RangeAction<?>> availableRangeActions = crac.getRangeActions().stream()
  .filter(ra -> RaoUtil.isRemedialActionAvailable(ra, automatonState, rangeActionSensitivity, crac.getFlowCnecs(), network, raoParameters))
  .collect(Collectors.toSet());
```

File: data/crac-io/crac-io-json/src/main/java/com/farao_community/farao/data/crac_io_json/deserializers/OnAngleConstraintArrayDeserializer.java [hunk 12]
```diff
@@ -24,13 +26,20 @@ public final class OnAngleConstraintArrayDeserializer {
     private OnAngleConstraintArrayDeserializer() {
     }
 
-    public static void deserialize(JsonParser jsonParser, RemedialActionAdder<?> ownerAdder) throws IOException {
+    public static void deserialize(JsonParser jsonParser, RemedialActionAdder<?> ownerAdder, String version) throws IOException {
+        boolean isUsageMethodPresent = false;
+        Instant instant = Instant.PREVENTIVE;
```
[the remark this answers is not in the document]
Comment (MartinBelthle) [reply data/crac-io/crac-io-json/src/main/java/com/farao_community/farao/data/crac_io_json/deserializers/OnAngleConstraintArrayDeserializer.java:31]: I need it to know if I'm in automatonState and therefore put UsageMethod.FORCED. I moved it into the case INSTANT.

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/InjectionSetPointActionCreationTest.java [hunk 13]
```diff
@@ -89,7 +89,7 @@ void testImportInjectionSetPointActions() {
         // on-state-included-curative-remedial-action (on state)
         NetworkAction ra3 = cracCreationContext.getCrac().getNetworkAction("on-state-included-curative-remedial-action");
         assertEquals("RA3", ra3.getName());
-        assertEquals(UsageMethod.FORCED, ra3.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra3.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/InjectionSetPointActionCreationTest.java:92]: for now this is supposed to stay FORCED
@bqth29 can confirm

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 14]
```diff
@@ -54,7 +54,7 @@ void testTC2ImportNetworkActions() {
         // RA13 (on state)
         NetworkAction ra13 = cracCreationContext.getCrac().getNetworkAction("1fd630a9-b9d8-414b-ac84-b47a093af936");
         assertEquals("RA13", ra13.getName());
-        assertEquals(UsageMethod.FORCED, ra13.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra13.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:57]: same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 15]
```diff
@@ -63,7 +63,7 @@ void testTC2ImportNetworkActions() {
         // RA22 (on state)
         NetworkAction ra22 = cracCreationContext.getCrac().getNetworkAction("d856a2a2-3de4-4a7b-aea4-d363c13d9014");
         assertEquals("RA22", ra22.getName());
-        assertEquals(UsageMethod.FORCED, ra22.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra22.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:66]: same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 16]
```diff
@@ -72,7 +72,7 @@ void testTC2ImportNetworkActions() {
         // RA14 (on state)
         NetworkAction ra14 = cracCreationContext.getCrac().getNetworkAction("c8bf6b19-1c3b-4ce6-a15c-99995a3c88ce");
         assertEquals("RA14", ra14.getName());
-        assertEquals(UsageMethod.FORCED, ra14.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra14.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:75]: same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 17]
```diff
@@ -81,7 +81,7 @@ void testTC2ImportNetworkActions() {
         // RA21 (on state)
         NetworkAction ra21 = cracCreationContext.getCrac().getNetworkAction("fb487cc2-0f7b-4958-8f66-1d3fabf7840d");
         assertEquals("RA21", ra21.getName());
-        assertEquals(UsageMethod.FORCED, ra21.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra21.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:84]: same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 18]
```diff
@@ -90,7 +90,7 @@ void testTC2ImportNetworkActions() {
         // RA3 (on state)
         NetworkAction ra3 = cracCreationContext.getCrac().getNetworkAction("5e401955-387e-45ce-b126-dd142b06b20c");
         assertEquals("RA3", ra3.getName());
-        assertEquals(UsageMethod.FORCED, ra3.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra3.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:93]: same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 19]
```diff
@@ -99,7 +99,7 @@ void testTC2ImportNetworkActions() {
         // RA5 (on state)
         NetworkAction ra5 = cracCreationContext.getCrac().getNetworkAction("587cb391-ed16-4a1d-876e-f90241addce5");
         assertEquals("RA5", ra5.getName());
-        assertEquals(UsageMethod.FORCED, ra5.getUsageRules().iterator().next().getUsageMethod());
+        assertEquals(UsageMethod.AVAILABLE, ra5.getUsageRules().iterator().next().getUsageMethod());
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:102]: same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java [hunk 20]
```diff
@@ -126,7 +126,7 @@ void testImportNetworkActions() {
 
         // RA4 (on state)
         assertTopologicalActionImported(cracCreationContext, "on-contingency-state-included-curative-topological-remedial-action", "RA4", "BBE1AA1  BBE4AA1  1");
-        assertHasOnContingencyStateUsageRule(cracCreationContext, "on-contingency-state-included-curative-topological-remedial-action", "bbda9fe0-77e0-4f8e-b9d9-4402a539f2b7", CURATIVE, UsageMethod.FORCED);
+        assertHasOnContingencyStateUsageRule(cracCreationContext, "on-contingency-state-included-curative-topological-remedial-action", "bbda9fe0-77e0-4f8e-b9d9-4402a539f2b7", CURATIVE, UsageMethod.AVAILABLE);
```
Comment (pet-mit) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/TopologicalActionCreationTest.java:129]: same

File: data/crac-io/crac-io-json/src/test/resources/retrocompatibility/v1/crac-v1.8.json
Comment (pet-mit) [inline data/crac-io/crac-io-json/src/test/resources/retrocompatibility/v1/crac-v1.8.json:1]: you mustn't change this old file, it should be importable thanks to retrocompatibility.
however, you should create a new v2.0 file with its tests

File: data/crac-io/crac-io-json/src/test/resources/retrocompatibility/v1/crac-v1.9.json
Comment (pet-mit) [inline data/crac-io/crac-io-json/src/test/resources/retrocompatibility/v1/crac-v1.9.json:1]: you mustn't change this old file, it should be importable thanks to retrocompatibility.
however, you should create a new v2.0 file with its tests

File: data/crac/crac-api/src/main/java/com/farao_community/farao/data/crac_api/RemedialAction.java [hunk 21]
```diff
@@ -55,18 +55,10 @@ public interface RemedialAction<I extends RemedialAction<I>> extends Identifiabl
...
-    boolean isRemedialActionAvailable(State state);
-
-    /**
-     * Evaluates if the remedial action is available depending on its UsageMethod.
-     */
-    boolean isRemedialActionAvailable(State state, boolean evaluatedCondition);
-
     Set<FlowCnec> getFlowCnecsConstrainingUsageRules(Set<FlowCnec> perimeterCnecs, Network network, State optimizedState);
 
+    Set<FlowCnec> getFlowCnecsConstrainingForOneUsageRule(UsageRule usageRule, Set<FlowCnec> perimeterCnecs, Network network);
```
Comment (pet-mit) [inline data/crac/crac-api/src/main/java/com/farao_community/farao/data/crac_api/RemedialAction.java:60]: I think these methods belong in RaoUtil (you will need to add the RA as an argument of the first). And they are pretty ugly IMO, they may well deserve a refactoring, but we can do that later

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/InjectionSetPointActionCreationTest.java (hunk 13 above)
Comment (MartinBelthle) [reply data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/InjectionSetPointActionCreationTest.java:92]: Indeed, but we thought TSOs probably don't understand what `FORCED` means and will change their specifications. 

We were supposed to have TSOs response on this subject last week but we didn't :confused:

Moreover, with this dev we ignore `FORCED` remedial actions other than for `Instant.AUTO` so it won't work :disappointed:

**Review by pet-mit — APPROVED** [review]

**Review by pet-mit — APPROVED** [review]
